src/realm/object-store/dictionary.hpp

```diff
@@ -55,6 +55,34 @@ class Dictionary : public object_store::Collection {
         m_dict->clear();
     }
 
+    Mixed get_any(StringData key)
+    {
+        return m_dict->get(key);
+    }
+    Mixed get_any(size_t ndx) const final
+    {
+        return m_dict->get_any(ndx);
+    }
+    bool try_get_any(StringData key, Mixed* result)
```

Comment:
Just a question, but is the reason we're not using `util::Optional<Mixed> try_get_any(StringData key)` as a signature that `Mixed<Optional>` in the sort-of public API has very ambiguous intent (as `Mixed` already has a null state)?

Reply:
My suggestion was directly translated from the C# API and I guess @jedelbo just went with it. I don't mind `Optional<Mixed>` if that's more natural.

Reply:
`util::Optional<Mixed>` would have worked too. It was just the other interface that was suggested.

Reply:
The function is now changed to return util::Optional<Mixed>